### PR TITLE
Transform `f1_accumdepr_prvsn` dbf + xbrl

### PR DESCRIPTION
**Pull request by zaneselvans**
## PR Description
* Mapped DBF rows to XBRL factoids for the `f1_accumdepr_prvsn` table.

## PR Checklist

Before requesting a review of your pull request, please make sure you've done the following:

* [x] Merge the most recent version of `dev`
* [x] Do your own review of the PR. Highlight areas where reviewers should focus their attention or you need help.
* [x] Write good docstrings for any new modules, classes, functions.
* [x] Ensure that new functions / methods / classes have at least basic unit tests.
* [x] Update the [release notes](https://catalystcoop-pudl.readthedocs.io/en/latest/release_notes.html).
* [x] (N/A) Make sure new analyses include defensive sanity checks that will catch unexpected data issues.
* [x] (N/A) Ensure that the full data validation tests pass locally.
* [x] Verify that all of the CI checks on your PR are passing.

### CODE REVIEW

File: src/pudl/extract/ferc1.py
Comment (zaneselvans) [inline src/pudl/extract/ferc1.py:177]: Not a huge fan of these table names, but also not sure what would be better... if you have suggestions!

File: src/pudl/metadata/fields.py
Comment (zaneselvans) [inline src/pudl/metadata/fields.py:469]: Left these commented out on purpose, just as an indicator of what is in there. Maybe should remove or make more comments.

File: src/pudl/package_data/ferc1/dbf_to_xbrl.csv
Comment (zaneselvans) [inline src/pudl/package_data/ferc1/dbf_to_xbrl.csv:511]: There are some weird mappings in here, see the notes.  E.g. the starting balance and 2 separate (hopefully identical) ending balances, all of which correspond to the one value reported in the instant XBRL table, because of the reshaping that has to happen to create distinct `starting_balance` and `ending_balance` and a desire to be able to distinguish between the separately reported `ending_balance` for functional classifications vs. plant/account sections.

File: src/pudl/package_data/ferc1/dbf_to_xbrl.csv
Comment (zaneselvans) [inline src/pudl/package_data/ferc1/dbf_to_xbrl.csv:2119]: Just changed the column that your notes were in (they were in the `dbf_only` column before)

File: src/pudl/transform/ferc1.py
Comment (zaneselvans) [inline src/pudl/transform/ferc1.py:185]: Changed b/c it can be more or less.

File: src/pudl/transform/ferc1.py
Comment (zaneselvans) [inline src/pudl/transform/ferc1.py:787]: just a formatting change here -- got rid of the trailing comma since it fits on one line and that's how the other similar calls are.

File: src/pudl/transform/ferc1.py
Comment (zaneselvans) [inline src/pudl/transform/ferc1.py:3001]: Not a fan of this embedded rename, but because of the order in which the `wide_to_tidy` and `rename_columns` calls happen, I couldn't do it in either the `xbrl` or `instant_xbrl` rename steps.  Is there a better way?

File: src/pudl/transform/ferc1.py
Comment (zaneselvans) [inline src/pudl/transform/ferc1.py:2943]: Are there other examples of "drop rows with these labels" in other tables that would justify making this into a (simple) parameterized method?

File: src/pudl/transform/params/ferc1.py
Comment (zaneselvans) [inline src/pudl/transform/params/ferc1.py:250]: I pulled out several string categorizations that show up in various places multiple times, but I didn't apply them outside of my own table yet. What do you think about switching over to using them where they apply?

File: src/pudl/transform/params/ferc1.py
Comment (zaneselvans) [inline src/pudl/transform/params/ferc1.py:3580]: Were you going to pull out the simple "there's only one category" case into its own function? Did it already happen and I missed it, or did you decide not to, or is it still in the works?

**Review by zaneselvans — COMMENTED** [review]
@cmgosnell left some pointers / questions / comments for you in this review.

Other questions:
* Should we drop rows that have NA value in the one data column (it's around half of all rows I think, since leased / future plant are rarely filled in).
* What metadata should get merged in or added by hand other than FERC Account? Only a minority of the rows included here correspond to FERC Accounts.

File: src/pudl/metadata/fields.py
Comment (cmgosnell) [reply src/pudl/metadata/fields.py:469]: yeah i would comment abt why if you want these here at all. I would personally not include them bc its pretty known that we should do some automatic enum-ing for all of the columns that get the `categorize_strings`

Also we've been using `{word_word}_type` for many of these category columns. I certainly won't die on the anti-classification hill, but I like the consistency of using a `type` suffix.

File: src/pudl/transform/ferc1.py
Comment (cmgosnell) [reply src/pudl/transform/ferc1.py:3001]: Since you are currently not doing an instant_xbrl rename, you could swap the order of `process_instant_xbrl`, so it would be:
* unstack start/end balance
* super

until you merge dev in since austin made this unstack a general transformer, you'll fully override `process_instant_xbrl` with:
* unstack
* rename

Comment (cmgosnell) [reply src/pudl/transform/ferc1.py:3001]: also is `ending_balance_accounts_electric_plant` supposed to be `ending_balance_electric_plant`?

File: src/pudl/transform/params/ferc1.py
Comment (cmgosnell) [reply src/pudl/transform/params/ferc1.py:3580]: we haven't done this yet. I still think its a good idea but not the highest priority as this is a fine option for now.

File: src/pudl/metadata/resources/ferc1.py
Comment (cmgosnell) [inline src/pudl/metadata/resources/ferc1.py:16]: I really think we should not propagate a ton of different column names that all refer to $ values. we have several instances of `{something}_value` which imo is better than this name. or `balance`, which i think would be appropriate in this table? or just `value`!

File: src/pudl/package_data/ferc1/dbf_to_xbrl.csv
Comment (cmgosnell) [reply src/pudl/package_data/ferc1/dbf_to_xbrl.csv:511]: i think i need to talk through this to understand what is going on here.

in general i think it is fine if these labels don't correspond to xbrl_factoid's per-say but do correspond to categories in the end table... but i'm a little confused about the layers of category columns in this table and how it relates to these mappings. these are all `ferc_account_label`'s... (which makes start/end balance a little weird). but these labels correspond to the stacked xbrl column stems? including start/end balance bc you invert the name in that process instant rename?

**Review by cmgosnell — CHANGES_REQUESTED** [review]
i'd love to talk some of these things over. all in all it looks good, but i have a few process suggestions.

I think my biggest concern is having these values cooresponding to annual differences in depreciation in the same column as `starting_balance` and `ending_balance`.

* RMI requested we keep all the nulls for the structure of the possible values. This is obviously fine and doable. I don't love it overall... especially for these tables that have more than one categorical column and thus many many nulls. Plus i believe it should be very easy for us to construct the possible categories given all the metadata. but at least until that point we should keep them.
* I've been adding the xbrl row type and the are you a credit or debit metadata

File: src/pudl/extract/ferc1.py
Comment (cmgosnell) [reply src/pudl/extract/ferc1.py:177]: it could be `utility_plant_depreciation_ferc1`

Comment (zaneselvans) [reply src/pudl/extract/ferc1.py:177]: What would the other tables be in relation to that -- the one organized by functional classification, and the one containing the unstructured data?

File: src/pudl/metadata/fields.py
Comment (zaneselvans) [reply src/pudl/metadata/fields.py:469]: Yeah maybe I should just delete

File: src/pudl/transform/ferc1.py
Comment (zaneselvans) [reply src/pudl/transform/ferc1.py:3001]: So you're suggesting redefining the whole `process_instant_xbrl` method, to do the same operations but in a different order?

File: src/pudl/transform/params/ferc1.py
Comment (zaneselvans) [reply src/pudl/transform/params/ferc1.py:250]: I thought we might concatenate the special "other" categories with a `dict | dict` in the specific params.

File: src/pudl/metadata/fields.py
Comment (cmgosnell) [reply src/pudl/metadata/fields.py:469]: or just `classification_type`

File: src/pudl/package_data/ferc1/dbf_to_xbrl.csv
Comment (zaneselvans) [reply src/pudl/package_data/ferc1/dbf_to_xbrl.csv:511]: Get rid of the  "account" in `ending_balance_account` column name in the account oriented table in the end (only meaningful in the context of the mapping, not the table)

File: src/pudl/transform/ferc1.py
Comment (zaneselvans) [reply src/pudl/transform/ferc1.py:3001]: Yes -- and use the `rename_columns` with instant_xbrl as the stage, but in a different order.

File: src/pudl/transform/ferc1.py
Comment (zaneselvans) [reply src/pudl/transform/ferc1.py:2943]: Hold off on this until #2155 gets merged.

File: src/pudl/transform/ferc1.py
Comment (zaneselvans) [reply src/pudl/transform/ferc1.py:2941]: Communicate more about WHY the `utility_type` should be added in here.

File: src/pudl/metadata/fields.py
Comment (zaneselvans) [reply src/pudl/metadata/fields.py:469]: Okay I went with `plant_classification_type` here just to be not utterly generic.

File: src/pudl/transform/ferc1.py
Comment (zaneselvans) [reply src/pudl/transform/ferc1.py:3001]: Implemented, fully overriding the`process_instant_xbrl()` method and using the `instant_xbrl` rename stage.

File: src/pudl/metadata/resources/ferc1.py
Comment (zaneselvans) [reply src/pudl/metadata/resources/ferc1.py:19]: I decided to use `utility_plant_value` here since that's what was used in the other depreciation table (`utility_plant_summary_ferc1`) that seems to be referring to the same value (in fact I think this table is aggregated up into that one)

File: src/pudl/package_data/ferc1/dbf_to_xbrl.csv
Comment (zaneselvans) [reply src/pudl/package_data/ferc1/dbf_to_xbrl.csv:511]: Removed this spurious "account" in the `instant_xbrl` rename.

Still not sure what these accounting categories should be called, since only a few of them are FERC accounts.

File: src/pudl/transform/ferc1.py
Comment (zaneselvans) [reply src/pudl/transform/ferc1.py:2941]: Okay, added a sentence explaining each of these actions.

File: src/pudl/transform/ferc1.py
Comment (zaneselvans) [reply src/pudl/transform/ferc1.py:2992]: Okay I'll be ready to integrate this when Austen's PR comes through.

**Comment by zaneselvans** [comment]
@cmgosnell I think I've addressed all the issues that we discussed. Please take another look! Hopefully the CI passes.

File: src/pudl/extract/ferc1.py
Comment (zaneselvans) [reply src/pudl/extract/ferc1.py]: I went with:
* `electric_plant_depreciation_changes_ferc1` and
* `electric_plant_depreciation_functional_ferc1`

And I guess we could do an `electric_plant_depreciation_changes_unstructured_ferc1` or something.  We should come up with a convention for the unstructured table names.

File: src/pudl/transform/ferc1.py
Comment (cmgosnell) [inline src/pudl/transform/ferc1.py:2992]: so you are renaming the original xbrl label as start bal, then effectively duplicating that record with all of its attributes and calling it ending balance because we need it twice. yeah that makes sense for this table imo

**Review by cmgosnell — CHANGES_REQUESTED** [review]
These look like great updates! My only remaining concern which i suppose isn't the biggest deal is the name `ferc_account_label`. This seems more like a `depreciation_type` or `depreciation_expense_type` or something, some of which are associated with ferc accounts.

File: src/pudl/transform/ferc1.py
Comment (zaneselvans) [reply src/pudl/transform/ferc1.py:2992]: It's not pretty, but that seemed to be what was needed for consistency.

File: src/pudl/transform/ferc1.py
Comment (zaneselvans) [inline src/pudl/transform/ferc1.py:3454]: People aren't remembering to update this, and I don't know if that means they think they're testing their new code, but aren't, or just aren't using it. But this way it will default to running all the tables with 2 years of data, even without being updated.

Comment (cmgosnell) [reply src/pudl/transform/ferc1.py:3454]: I have been running the full ferc etl like this in a notebook before running the full tests. regardless, this is a good idea bc it has been hard to remember all of the little places to update these things.

File: src/pudl/transform/ferc1.py
Comment (cmgosnell) [inline src/pudl/transform/ferc1.py:505]: what a helpful feature!

File: src/pudl/transform/ferc1.py
Comment (cmgosnell) [inline src/pudl/transform/ferc1.py:1066]: I think this is generally fine! and certainly good enough to merge in. In fact I think  we should!! 

but also I'm curious if instead of adding this as an argument if we can add it to `params` within this method. iirc you (reasonably!) made the params base class immutable. Which i think is the correct default. There are several instances of parameter elements that we could generate within the class itself which would enable us to stop hard coding things like the source table.

**Review by cmgosnell — APPROVED** [review]
🦖 🎉

**Pull request merged**
